README


This archive contains the distribution Dist-Zilla-Plugin-MakeMaker-Fallback,
version 0.016:

  Generate a Makefile.PL containing a warning for legacy users

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.025.



Changes
Revision history for Dist-Zilla-Plugin-MakeMaker-Fallback

0.016     2014-11-12 05:36:19Z
          - bump up prereq version for [MakeMaker::Awesome] to ensure needed
            template variables are made available

0.015     2014-11-12 03:53:50Z
          - remove configure requires declarations added for Makefile.PL
            (technically they are only needed when configure requires is
            already not being respected)

0.014     2014-10-29 02:44:49Z
          - document the double testing behaviour that changed with version
            0.007 (thanks for noticing, Apocalypse!)
          - remove perl from the list of configure requires prereqs to check
            for in Makefile.PL

0.013     2014-08-16 00:02:55Z
          - fix test from previous release's fix that was incompatible with
            earlier versions of [MakeMaker]'s practice of always setting
            eumm_version

0.012     2014-08-14 02:03:30Z
          - fix munging of "use ExtUtils::MakeMaker <version>" since
            [MakeMaker::Awesome] version 0.22 changed the template
          - lower minimum version of perl required by removing a use of \N in
            a regular expression

0.011     2014-08-07 02:27:53Z
          - properly include %configure_requires content in Makefile.PL
            (broken under Dist::Zilla::Plugin::MakeMaker::Awesome 0.23,
            released 2014-07-30).

0.010     2014-06-07 06:29:43Z
          - print a useful log message during dzil test --verbose
          - "use ExtUtils::MakeMaker <version>" now stripped out of
            Makefile.PL, to allow non-configure-requires-compliant
            installers to still proceed

0.009     2014-03-24 03:52:26Z
          - fix test which should be skipped when an optional prereq is not
            present

0.008     2014-03-23 02:43:22Z
          - properly dump configs by resolving silent role composition sub
            clobbering issue

0.007     2014-03-09 05:35:08Z
          - switch to renamed form of now-deprecated plugin
            [MungeData::WithDataSection]
          - "dzil test" will run tests without extra testing variables, for
            "dzil release" or "dzil test --release" -- this will weed out test
            issues that only manifest when AUTHOR_TESTING or RELEASE_TESTING
            are *not* set (e.g. bad test count, conditional module loading).

0.006     2014-01-25 03:52:42Z
          - now sorting the dumped %configure_requires prereqs, for
            minimizing diffs between releases

0.005     2013-12-14 17:25:20Z
          - warnings can now be bypassed entirely with the
            PERL_MM_FALLBACK_SILENCE_WARNING environment variable
          - now also mentioning the relevant CPAN.pm/CPANPLUS configuration
            options

0.004     2013-11-20 21:49:00Z
          - tighten up the interactive terminal detection logic

0.003     2013-10-15 01:26:28Z
          - only display the warning if configure_require prereqs are
            genuinely missing (the user may have explicitly chosen to run
            Makefile.PL) (RT#89434)

0.002     2013-09-25 02:15:24Z
          - bump up dependency on [MakeMaker::Awesome] to avoid missing lib
            errors

0.001     2013-09-22 21:33:34Z
          - Initial release.



Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-MakeMaker-Fallback was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.015
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.009005;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Generate a Makefile.PL containing a warning for legacy users",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-MakeMaker-Fallback",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.009005",
  "NAME" => "Dist::Zilla::Plugin::MakeMaker::Fallback",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::MakeMaker::Awesome" => "0.15",
    "Dist::Zilla::Role::AfterBuild" => 0,
    "List::Util" => 0,
    "Moose" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
    "Dist::Zilla::Role::InstallTool" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "Test::Requires" => 0,
    "if" => 0
  },
  "VERSION" => "0.016",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Capture::Tiny" => 0,
  "Dist::Zilla::Plugin::MakeMaker::Awesome" => "0.15",
  "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::InstallTool" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "List::Util" => 0,
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "Test::Requires" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-MakeMaker-Fallback was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.012
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-MakeMaker-Fallback was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.009005;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.025.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.009005"
      },
      "dist_abstract" => "Generate a Makefile.PL containing a warning for legacy users",
      "dist_author" => [
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-MakeMaker-Fallback",
      "dist_version" => "0.016",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::MakeMaker::Fallback",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Dist::Zilla::Plugin::MakeMaker::Awesome" => "0.15",
        "Dist::Zilla::Role::AfterBuild" => 0,
        "List::Util" => 0,
        "Moose" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.009005",
        "strict" => 0,
        "version" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "Capture::Tiny" => 0,
        "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
        "Dist::Zilla::Role::InstallTool" => 0,
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "Test::Requires" => 0,
        "if" => 0,
        "perl" => "5.009005"
      }
    );


    my %fallback_build_requires = (
      "Capture::Tiny" => 0,
      "Dist::Zilla::Plugin::ModuleBuildTiny" => 0,
      "Dist::Zilla::Role::InstallTool" => 0,
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "Test::Requires" => 0,
      "if" => 0,
      "perl" => "5.009005"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
